### DISCUSSION TITLE
Fetching variables from HTTP-Post with microdot-Server

### DISCUSSION THREAD

**Discussion by takkowa** [micropython/micropython#13112] [Using MicroPython]
Hi,

I'm trying to fetch some data from a a website, that a user has enetered before.
I modified the microdot example for templates, but it seems that I got something wrong.

The Code runs without any errors, but the variables, witch should contain the user entered values (wlanSsid & wlanPsk)  stay null.

I uesed tzhe following code:

```
import AutoWlan  # connects automatically to wlan

from microdot.microdot import Microdot, Response
from microdot.microdot_utemplate import render_template

app = Microdot( )
Response.default_content_type = 'text/html'


@app.route('/', methods=['GET', 'POST'])
def index(req):
    wlanSsid = None
    wlanPsk = None
    if req.method == 'POST':
        wlanSsid = req.form.get('wlanSsid')
        wlanPsk = req.form.get('wlanPsk')
        print(wlanSsid)
        print(wlanPsk)
    idxHtm = render_template('index.html', wlanSsid='wlanSsid', wlanPsk='wlanPsk')
    return idxHtm

app.run(port = 80 )

```

and the related index.html:
```
{% args wlanSsid, wlanPsk %}
<!doctype html>
<html>
  <head>
    <title>Enter credentials</title>
    <meta charset="UTF-8">
  </head>
  <body>
    <h1>Enter WLAN credentials</h1>
    <form method="POST">
      <p>
        Enter WLAN credentials? <br /> <br />
          WLAN SSID : <input type="text" wlanSsid="wlanSsid" autofocus /> <br /> <br />
          WLAN password : <input type="text" wlanPsk="wlanPsk" autofocus /><br /> <br />
      </p>
      <input type="submit" value="Submit" />
    </form>
  </body>
</html>
```

Do you have any idea, what I did wrong?

**Comment by sophiedegran** [comment]
Hi takkowa
I think your forgot name and id on your html input text .. that's need in html post (sure id)

**Comment by sophiedegran** [comment]
So it will be 
wlanSsid = req.form [wlanSsid] 
for input type="text" name="wlanSsid" id="wlanSsid'....

**Comment by takkowa** [comment]
Hi Sophiedegran,

thank you for your hints, but it did not really help:

In your first message, you said, that name and id is missing. I don't know if I got you right (probably not :-)) but I deleted the original parameter "name" form the microdot examnple by replacing it with wlanSsid and wlanPsk. IMO "name" is in the original microdot example the name of a variable, wich contains the name entered by the user.  I did not understand, what you meant with "missing id" - what id do you mean?

In your second message you suggest to replace the statement wlanSsid = req.form.get('wlanSsid') with "wlanSsid = req.form[wlanSsid]".
trying that gives me an keyerror. (shouldn't it be req.form['wlanSsid'] - gives also an key-Error) - what makes the difference between the get method an the access with the "[ ]" brackets?
Eventually your hints point into the right direction: If I print req.form it is empty! There should probably some values in it ...
Any other ideas?

**Reply by sophiedegran** [reply]
Yes do simply req.form.get ("wlanSsid")
If you want to know what you 're post do simply print (req.headers) in microdot
It's name that you have to inform on post .. (I made a mistake as I use id for javascript sorry)
